client/src/http_default.rs

```diff
@@ -134,46 +108,81 @@ impl RequestBuilder for DefaultRequestBuilder {
         K: AsRef<str>,
         V: ToString,
     {
-        Self {
-            params: Some(
-                params
-                    .into_iter()
-                    .map(|pair| {
-                        let (k, v) = pair.borrow();
-                        (k.as_ref().to_owned(), v.to_string())
-                    })
-                    .collect(),
-            ),
-            ..self
-        }
+        self.and_then(|b| Ok(b.params(params)))
+    }
+
+    fn body(self, data: Option<Vec<u8>>) -> Self::Output {
```

Comment:
`Option<Vec<u8>>` → `Vec<u8>`. Fewer lines of code, and every implicit `None` that can change the behaviour, have an explicit `Vec::new`.